src/opensles/AudioInputStreamOpenSLES.cpp

```diff
@@ -234,41 +240,64 @@ Result AudioInputStreamOpenSLES::setRecordState(SLuint32 newState) {
 }
 
 Result AudioInputStreamOpenSLES::requestStart() {
-
-    LOGD("AudioInputStreamOpenSLES::requestStart()");
+    LOGD("AudioInputStreamOpenSLES(): %s() called", __func__);
+    mLock.lock();
     StreamState initialState = getState();
-    if (initialState == StreamState::Closed) return Result::ErrorClosed;
+    switch (initialState) {
+        case StreamState::Starting:
+        case StreamState::Started:
+            mLock.unlock();
+            return Result::OK;
+        case StreamState::Closed:
+            mLock.unlock();
+            return Result::ErrorClosed;
+        default:
+            break;
+    }
 
+    setDataCallbackEnabled(true);
```

Comment:
As previous comment. Not sure why the stream should always be using callbacks.

Reply:
Enabled just means it is not disabled. It does not mean it will be used. But that is confusing. I will clarify or modify the code.

Reply:
I cleaned up the enable and also refactored requestStart for OpenSL ES.